pkg/runner/step_context.go

```diff
@@ -220,6 +221,38 @@ func (sc *StepContext) setupAction(actionDir string, actionPath string) common.E
 		if os.IsNotExist(err) {
 			f, err = os.Open(filepath.Join(actionDir, actionPath, "action.yaml"))
 			if err != nil {
+				if sc.Step.With != nil {
+					if val, ok := sc.Step.With["args"]; ok {
+						trampoline, err2 := Asset("trampoline.js")
+						if err2 != nil {
+							return err
+						}
+						err2 = ioutil.WriteFile(filepath.Join(actionDir, actionPath, "trampoline.js"), trampoline, 0755)
+						if err2 != nil {
+							return err
+						}
+						sc.Action = &model.Action{
+							Name: "(Synthetic)",
+							Inputs: map[string]model.Input{
+								"cwd": model.Input{
+									Description: "(Actual working directory)",
+									Required:    false,
+									Default:     filepath.Join(actionDir, actionPath),
+								},
+								"command": model.Input{
+									Description: "(Actual program)",
+									Required:    false,
+									Default:     val,
+								},
+							},
+							Runs: model.ActionRuns{
+								Using: "node12",
+								Main:  "trampoline.js",
```

Comment:
Is there a better place to stash this file? It feels risky to write into the file system of an action...
Alternatively, is there a suggested way to uniquely/randomly generate a directory name?